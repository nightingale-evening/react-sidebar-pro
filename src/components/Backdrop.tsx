import React from 'react';
import styled, { CSSObject } from 'styled-components';
import { sidebarClasses } from '../utils/utilityClasses';

interface BackdropProps extends React.HTMLAttributes<HTMLDivElement> {
  rootStyles?: CSSObject;
}

const StyledBackdrop = styled.div<Pick<BackdropProps, 'rootStyles'>>`
  position: fixed;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  z-index: 1;
  background-color: rgb(0, 0, 0, 0.3);

  ${({ rootStyles }) => rootStyles}
`;

export const Backdrop: React.FC<BackdropProps> = ({ rootStyles, ...rest }) => {
  return (
    <StyledBackdrop
      data-testid={`${sidebarClasses.backdrop}-test-id`}
      role="button"
      tabIndex={0}
      aria-label="backdrop"
      rootStyles={rootStyles}
      {...rest}
    />
  );
};
