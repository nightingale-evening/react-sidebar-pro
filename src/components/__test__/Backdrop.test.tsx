import React from 'react';
import { customRender, screen } from '../../utils/testUtils';
import { sidebarClasses } from '../../utils/utilityClasses';
import { Backdrop } from '../Backdrop';

describe('Backdrop', () => {
  it('should initialize Backdrop correctly', async () => {
    customRender(<Backdrop />);
    const backdropElem = screen.getByTestId(`${sidebarClasses.backdrop}-test-id`);

    expect(backdropElem).toBeInTheDocument();
    expect(backdropElem).toHaveStyle({
      position: 'fixed',
      top: '0px',
      right: '0px',
      bottom: '0px',
      left: '0px',
      'background-color': 'rgb(0, 0, 0, 0.3)',
      'z-index': '1',
    });
  });

  it('should render rootStyles', async () => {
    customRender(
      <Backdrop
        rootStyles={{
          backgroundColor: 'red',
          opacity: 0.5,
        }}
      />,
    );
    const backdropElem = screen.getByTestId(`${sidebarClasses.backdrop}-test-id`);
    expect(backdropElem).toHaveStyle({
      'background-color': 'red',
      opacity: 0.5,
    });
  });
});
