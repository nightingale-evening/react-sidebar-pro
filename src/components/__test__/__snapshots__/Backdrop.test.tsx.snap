// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop basic snapshot  1`] = `
<div
  aria-label="backdrop"
  class="sc-bcXHqe jeMTJc"
  data-testid="ps-sidebar-backdrop-test-id"
  role="button"
  tabindex="0"
/>
`;
